src/OrchardCore.Modules/OrchardCore.Localization/Views/CulturePickerContainer.cshtml

```diff
@@ -0,0 +1,22 @@
+@using OrchardCore.Localization
+
+@{
+    var selectedCultureName = Model.CurrentCulture.Name;
+}
+<select id="culture" name="culture" class="form-select me-3" onchange="changeCulture()">
+  @foreach (var culture in Model.SupportedCultures)
+  {
+    <option selected="@(culture.Name == selectedCultureName ? "selected" : null)" value="@culture.Name">@culture.DisplayName</option>
```

Comment:
Changing this doesn't persist:

https://user-images.githubusercontent.com/1976647/179827705-54b56a69-80bf-4aea-91fc-b470b71ed223.mp4

Reply:
That issue was happen before the commit https://github.com/OrchardCMS/OrchardCore/pull/12017/commits/b9c14f87737c261cd8e058cf2caf7bd03bf67b94

Did you try it on the latest changes, I will share a screenshot from my local. One more thing you need check, please remove the old cookie if persist in your browser, becuase it might causes such issue

Reply:
I'm using the latest code and this happens with a vanilla browser too.

Reply:
I faced the exact same issue when I ran it before the commit https://github.com/OrchardCMS/OrchardCore/commit/b9c14f87737c261cd8e058cf2caf7bd03bf67b94 then I removed the old cookie everything works fine

Let me check it again and come back to you

Reply:
I'm on commit fba62199da9ba224b920428a9adde7d63ae8b55a.

Reply:
Works now, thank you.